daemon/api.go

```diff
@@ -196,19 +196,30 @@ func loginUser(c *Command, r *http.Request) Response {
 	}
 
 	discharge, err := store.DischargeAuthCaveat(loginData.Username, loginData.Password, macaroon, loginData.Otp)
-	if err == store.ErrAuthenticationNeeds2fa {
+	switch err {
+	case store.ErrAuthenticationNeeds2fa:
 		twofactorRequiredResponse := &resp{
 			Type: ResponseTypeError,
 			Result: &errorResult{
 				Kind:    errorKindTwoFactorRequired,
-				Message: store.ErrAuthenticationNeeds2fa.Error(),
+				Message: err.Error(),
 			},
 			Status: http.StatusUnauthorized,
 		}
 		return SyncResponse(twofactorRequiredResponse, nil)
-	}
-	if err != nil {
+	case store.Err2faFailed:
+		return SyncResponse(&resp{
+			Type: ResponseTypeError,
+			Result: &errorResult{
+				Kind:    errorKindTwoFactorFailed,
+				Message: err.Error(),
+			},
+			Status: http.StatusForbidden,
```

Comment:
401 is basically "(more) credentials needed" whereas 403 is "bad credentials", which is the difference between these two cases